seaworthy/tests-core/test_client.py

```diff
@@ -0,0 +1,220 @@
+import unittest
+
+import responses
+
+from seaworthy.checks import docker_client, dockertest
+from seaworthy.client import ContainerClient
+from seaworthy.containers.base import ContainerBase
+from seaworthy.dockerhelper import DockerHelper, fetch_images
+
+
+# Small (<4MB) image that echoes HTTP requests and runs without configuration
+IMG = 'jmalloc/echo-server'
+
+
+@dockertest()
+def setUpModule():  # noqa: N802 (The camelCase is mandated by unittest.)
+    with docker_client() as client:
+        fetch_images(client, [IMG])
+
+
+class DummySession:
+    def __init__(self):
+        self.requests = []
+        self.closes = 0
+
+    def request(self, *args, **kwargs):
+        self.requests.append((args, kwargs))
+
+    def close(self):
+        self.closes += 1
+
+
+class DummySessionFactory:
+    def __init__(self):
+        self.sessions = []
+
+    def __call__(self):
+        session = DummySession()
+        self.sessions.append(session)
+        return session
+
+
+class TestContainerClient(unittest.TestCase):
+    def make_helper(self):
+        dh = DockerHelper()
+        self.addCleanup(dh.teardown)
+        dh.setup()
+        return dh
+
+    @responses.activate
+    def test_defaults(self):
+        """
+        When the container client is configured with a host address and port,
+        requests are made against that address and port.
+        """
+        client = ContainerClient('127.0.0.1', '12345')
+
+        responses.add(responses.GET, 'http://127.0.0.1:12345/', status=200)
+        response = client.request('GET', [])
+
+        self.assertEqual(response.status_code, 200)
+
+        self.assertEqual(len(responses.calls), 1)
+        [call] = responses.calls
+        self.assertEqual(call.request.url, 'http://127.0.0.1:12345/')
+
+    @responses.activate
+    def test_url_defaults(self):
+        """
+        When the container client is configured with a host address and port,
+        and some URL defaults are set, requests are made agains that address
```

Comment:
"agains"? :-P

I think "[...] requests are made *to* that address [...]" would be better anyway.